xchange-stream-binance/src/main/java/info/bitrich/xchangestream/binance/BinanceStreamingMarketDataService.java

```diff
@@ -42,36 +47,20 @@
 import org.slf4j.LoggerFactory;
 
 public class BinanceStreamingMarketDataService implements StreamingMarketDataService {
-  private static final Logger LOG =
-      LoggerFactory.getLogger(BinanceStreamingMarketDataService.class);
-
-  private static final JavaType TICKER_TYPE =
-      getObjectMapper()
-          .getTypeFactory()
-          .constructType(
-              new TypeReference<
-                  BinanceWebsocketTransaction<TickerBinanceWebsocketTransaction>>() {});
-  private static final JavaType TRADE_TYPE =
-      getObjectMapper()
-          .getTypeFactory()
-          .constructType(
-              new TypeReference<
-                  BinanceWebsocketTransaction<TradeBinanceWebsocketTransaction>>() {});
-  private static final JavaType DEPTH_TYPE =
-      getObjectMapper()
-          .getTypeFactory()
-          .constructType(
-              new TypeReference<
-                  BinanceWebsocketTransaction<DepthBinanceWebSocketTransaction>>() {});
+  private static final Logger LOG = LoggerFactory.getLogger(BinanceStreamingMarketDataService.class);
+
+  private static final JavaType TICKER_TYPE = getTickerType();
+  private static final JavaType TRADE_TYPE = getTradeType();
+  private static final JavaType DEPTH_TYPE = getDepthType();
```

Comment:
not sure why this is getting modified but if you want to reduce lines of code try 

```
  private static final JavaType TICKER_TYPE = BinanceStreamingMarketDataService.<TickerBinanceWebsocketTransaction>getTypeObject();
  private static final JavaType TRADE_TYPE = BinanceStreamingMarketDataService.<TradeBinanceWebsocketTransaction>getTypeObject();
  private static final JavaType DEPTH_TYPE = BinanceStreamingMarketDataService.<DepthBinanceWebSocketTransaction>getTypeObject();
```

```
  private static <T> JavaType getTypeObject() {
    return getObjectMapper()
            .getTypeFactory()
            .constructType(new TypeReference<BinanceWebsocketTransaction<T>>() {});
  }
```

Reply:
Tried this. Got error. Seams type got erased.

`10:19:58.887 [nioEventLoopGroup-2-1] ERROR info.bitrich.xchangestream.binance.BinanceManualExample - ERROR in getting order book: 
java.lang.ClassCastException: class java.util.LinkedHashMap cannot be cast to class info.bitrich.xchangestream.binance.dto.DepthBinanceWebSocketTransaction (java.util.LinkedHashMap is in module java.base of loader 'bootstrap'; info.bitrich.xchangestream.binance.dto.DepthBinanceWebSocketTransaction is in unnamed module of loader 'app')
	at io.reactivex.internal.operators.observable.ObservableFilter$FilterObserver.onNext(ObservableFilter.java:46)
	at io.reactivex.internal.operators.observable.ObservableMap$MapObserver.onNext(ObservableMap.java:62)
	at io.reactivex.internal.operators.observable.ObservableMap$MapObserver.onNext(ObservableMap.java:62)
	at io.reactivex.internal.operators.observable.ObservableRefCount$RefCountObserver.onNext(ObservableRefCount.java:227)
	at io.reactivex.internal.operators.observable.ObservablePublishAlt$PublishConnection.onNext(ObservablePublishAlt.java:177)
	at io.reactivex.internal.observers.DisposableLambdaObserver.onNext(DisposableLambdaObserver.java:58)
	at io.reactivex.internal.operators.observable.ObservableCreate$CreateEmitter.onNext(ObservableCreate.java:66)
	at info.bitrich.xchangestream.service.netty.NettyStreamingService.handleChannelMessage(NettyStreamingService.java:487)`